tracks/bash/exercises/hamming/mentoring.md

```diff
@@ -0,0 +1,147 @@
+### Reasonable Solutions
+
+```bash
+#!/usr/bin/env bash
+main () {
+  (( "$#" == 2 )) || {
+      echo "Usage: hamming.sh <string1> <string2>"
+      exit 1
+  }
+  
+  # Regular vars are easier to read when doing fancy parameter expansion.
+  a=$1 b=$2 
+  
+  (( "${#a}" == "${#b}" )) || {
+      echo "left and right strands must be of equal length"
+      exit 1
+  }
+
+  declare -i count
+  for (( i = 0; i < "${#a}"; i++ )); do
+    [[ "${a:i:1}" == "${b:i:1}" ]] || count+=1
+  done
+
+  printf '%d' "$count"
```

Comment:
Output ought to have a trailing newline. It doesn't matter inside unit tests that string trailing newlines but if you try actually running the script, it becomes painfully obvious that it's not user friendly.

Reply:
Is the following the expected change?
printf '%d\n' "$count"

also why printf vs echo ?

Reply:
Yes. `printf "%d\n"` does the job.

http://wooledge.org/~greybot/meta/echo - last line

Reply:
## Do no options when using echo

"Never use options to echo! Ever! Any time you feel tempted to use echo -e, -n, or any other special feature of echo, use printf instead! If portability is a requirement, you should consider using printf exclusively and just ignore that echo even exists."
~https://wiki.bash-hackers.org/commands/builtin/echo 👍